Common/include/CConfig.hpp

```diff
@@ -108,6 +108,13 @@ class CConfig {
   unsigned short FFD_Blending;    /*!< \brief Kind of FFD Blending function. */
   su2double* FFD_BSpline_Order;   /*!< \brief BSpline order in i,j,k direction. */
   su2double FFD_Tol;              /*!< \brief Tolerance in the point inversion problem. */
+  bool FFD_IntPrev;               /*!< \brief Enables self-intersection prevention procedure within the FFD box. */
+  unsigned short FFD_IntPrev_Iter; /*!< \brief Amount of iterations for FFD box self-intersection prevention procedure. */
+  unsigned short FFD_IntPrev_Depth; /*!< \brief Maximum recursion depth for FFD box self-intersection procedure. */
+  bool ConvexityCheck;            /*!< \brief Enables convexity check on all mesh elements. */
+  unsigned short ConvexityCheck_Iter; /*!< \brief Amount of iterations for convexity check in deformations. */
+  unsigned short ConvexityCheck_Depth; /*!< \brief Maximum recursion depth for convexity check in deformations.*/
```

Comment:
You can define a struct or use a tuple with these 3 fields to make the code more concise, that way you would only need 2 "Gets" (one for Intersections the other for convexity).